docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
+like https://github.com/wg/wrk[wrk] and run the following command to send requests to the application.
++
+[source,bash]
+----
+./wrk -t1 -c 100 -d10s http://localhost:7000
+----
++
+This command results in roughly 8k requests per
+second, and the equivalent of log lines are also written.
+
+[discrete]
+== Step 3: View logs in Kibana
+
+. Log into Kibana and select the *Discover* app.
++
+There is a summary of the documents at the top, but let’s take a look at a single document.
++
+image:./images/monitor-java-app-kibana-single-document.png[Kibana single document view]
++
+You can see that a lot more data is indexed than just the event. There is information about
+the offset in the file, information about the component shipping the logs, the name of the shipper
+in the output, and there is a `message` field which contains log line contents.
++
+You can see there is a flaw in the request logging. If the user agent is `null`,
+something other than `null` is returned. Reading our logs is
+crucial; however, just indexing them gains us nothing.  To fix this, here is a new request logger.
++
+[source,java]
+----
+Javalin app = Javalin.create(config -> {
+    config.requestLogger((ctx, executionTimeMs) -> {
+        String userAgent = ctx.userAgent() != null ? ctx.userAgent() : "-";
+        logger.info("{} {} {} {} \"{}\" {}",
+                ctx.method(), ctx.req.getPathInfo(), ctx.res.getStatus(),
+                ctx.req.getRemoteHost(), userAgent, executionTimeMs.longValue());
+    });
+});
+----
++
+You may also want to fix this in the logging message in the main handler. And to speed things up,
+you'll want to call `ctx.userAgent()` only once.
++
+. Now let's have a look at the Logs app in {kib}. Select *Observability* -> *Logs*.
++
+If you want to see the streaming feature at work, run the following curl request in
+a loop while sleeping.
++
+[source,bash]
+----
+while $(sleep 0.7) ; do curl localhost:7000 ; done
+----
++
+. To view a continuous stream of log messages, click *Stream live*. You can also
+highlight specific terms, as shown here.
++
+image:./images/monitor-java-app-kibana-streaming.png[Kibana Logs UI Streaming]
++
+Looking at one of the documents being indexed, you can see that the log message
+is contained in a single field. Verify this by looking at one of those documents.
++
+[source,console]
+----
+GET filebeat-*/_search
+{
+  "size": 1
+}
+----
++
+Things to note:
+
+* When you compare the `@timestamp` field with the timestamp of the log message, you will
+notice that it differs. This means that when filtering based on the
+`@timestamp` field, you do not get the results you expect. The current
+`@timestamp` field reflects the timestamp of when the event was created within
+{filebeat}, not the timestamp of when the log event occurred in the
+application.
+* The log level cannot be filtered, and the same applies to the class name.
+
+[discrete]
+==== Structure logs
```

Comment:
Not sure this really belongs under "View logs in Kibana". I wonder if you need a separate step to contain the info about structuring logs, parsing exceptions, configuring log rotations, ingest node, and writing logs as JSON? Seems like a lot to put under "View logs in Kibana".

Reply:
I agree. I grouped the sections under an additional step: Work with your logs.